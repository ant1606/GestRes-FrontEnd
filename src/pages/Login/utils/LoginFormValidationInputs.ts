import { type LoginFormData, type ValidationInputMessage } from '../index.types';

export const validateUserEmail = (values: LoginFormData): ValidationInputMessage => {
  const emailToValidate = values.email.trim();
  const regExFormatEmail = /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/;
  const emailValidFormat = regExFormatEmail.test(emailToValidate);
  if (emailToValidate.length === 0) {
    return 'Debe ingresar el email del usuario';
  }
  if (!emailValidFormat) {
    return 'Formato incorrecto ingresado del email';
  }
  return null;
};

export const validateUserPassword = (values: LoginFormData): ValidationInputMessage => {
  const passwordToValidate = values.password.trim();
  if (passwordToValidate.length === 0) {
    return 'Debe ingresar el password del usuario';
  }
  return null;
};
