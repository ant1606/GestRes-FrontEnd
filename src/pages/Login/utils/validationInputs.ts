import { type LoginFormData } from '../index.types';

export const validateUserEmail = (values: LoginFormData): ValidationInputResult => {
  const emailToValidate = values.email.trim();
  const regExFormatEmail = /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/;
  const emailValidFormat = regExFormatEmail.test(emailToValidate);
  if (emailToValidate.length === 0) {
    return 'Debe ingresar el email del usuario';
  }
  if (!emailValidFormat) {
    return 'Formato incorrecto ingresado del email';
  }
  return null;
};

export const validateUserPassword = (values: LoginFormData): ValidationInputResult => {
  const passwordToValidate = values.password.trim();
  if (passwordToValidate.length === 0) {
    return 'Debe ingresar el password del usuario';
  }
  return null;
};
