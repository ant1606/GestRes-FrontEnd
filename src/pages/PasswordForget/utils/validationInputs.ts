import { type PasswordForgetFormData } from '../index.types';

export const validateUserEmail = (values: PasswordForgetFormData): ValidationInputResult => {
  const emailToValidate = values.email.trim();
  const regExFormatEmail = /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/;
  const emailValidFormat = regExFormatEmail.test(emailToValidate);
  if (emailToValidate.trim().length === 0) {
    return 'Debe ingresar el email del usuario';
  }
  if (!emailValidFormat) {
    return 'Formato incorrecto ingresado del email';
  }
  return null;
};
