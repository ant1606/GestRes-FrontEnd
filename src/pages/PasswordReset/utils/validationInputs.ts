import { type ResetPasswordFormData } from '../index.types';

export const validateUserEmail = (values: ResetPasswordFormData): ValidationInputResult => {
  const emailToValidate = values.email.trim();
  const regExFormatEmail = /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/;
  const emailValidFormat = regExFormatEmail.test(emailToValidate);
  if (emailToValidate.trim().length === 0) {
    return 'Debe ingresar el email del usuario';
  }
  if (!emailValidFormat) {
    return 'Formato incorrecto ingresado del email';
  }
  return null;
};

export const validateUserPassword = (values: ResetPasswordFormData): ValidationInputResult => {
  const passwordToValidate = values.password?.trim();
  if (passwordToValidate.trim().length === 0) {
    return 'Debe ingresar el password del usuario';
  }
  return null;
};

export const validateUserPasswordConfirmation = (
  values: ResetPasswordFormData
): ValidationInputResult => {
  const passwordConfirmationToValidate = values.passwordConfirmation?.trim();
  const passwordToValidate = values.password?.trim();
  if (passwordConfirmationToValidate.trim().length === 0) {
    return 'Debe ingresar la confirmación de password del usuario';
  }
  if (passwordConfirmationToValidate !== passwordToValidate) {
    return 'Las contraseñas ingresadas no son iguales';
  }
  return null;
};
