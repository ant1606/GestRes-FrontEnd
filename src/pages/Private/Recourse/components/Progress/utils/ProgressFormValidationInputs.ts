import moment from 'moment';
import { type ProgressFormData } from '../indext.types';

export const validateAdvancedAmount = (values: ProgressFormData): ValidationInputResult => {
  if (values.advanced < 1) {
    return 'El Avance  realizado debe ser mayor a 0';
  }
  if (values.lastProgress !== undefined && values.advanced <= values.lastProgress.advanced) {
    return `El Avance  realizado debe ser mayor al anterior avance realizado que fue  ${values.lastProgress.advanced}`;
  }
  if (values.lastProgress !== undefined && values.advanced > values.lastProgress.total) {
    return `Avance debe ser menor o igual a la cantidad Total del Recurso que es ${values.lastProgress.total}`;
  }
  return null;
};

export const validateDoneAmount = (values: ProgressFormData): ValidationInputResult => {
  // if (values.done < 1) {
  //   return 'Avance debe ser mayor a 0';
  // }
  // if (values.lastProgress !== undefined && values.done > values.lastProgress.pending) {
  //   return 'Avance debe ser menor a Pendiente';
  // }
  return null;
};

export const validateDate = (values: ProgressFormData): ValidationInputResult => {
  const myDate = moment(values.date);
  const dateOfLastStatus = moment(values.lastProgress?.date);
  const differenceBetweenDays = myDate.diff(dateOfLastStatus, 'days');
  if (differenceBetweenDays < 0)
    return 'No puede registrar una fecha anterior a la Ãºltima fecha del registro de estado del recurso';
  return null;
};

export const validateComment = (): ValidationInputResult => {
  return null;
};
