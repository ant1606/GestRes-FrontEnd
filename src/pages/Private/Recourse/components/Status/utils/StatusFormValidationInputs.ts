import moment from 'moment';
import { type StatusFormData } from '../index.types';
import { GLOBAL_STATUS_RECOURSE } from '@/config/globalConstantes';

export const validateFecha = (values: StatusFormData): ValidationInputResult => {
  console.log(values);
  const myDate = moment(values.date);
  const dateOfLastStatus = moment(values.lastStatusOfRecourse?.date);
  const differenceBetweenDays = myDate.diff(dateOfLastStatus, 'days');
  if (differenceBetweenDays < 0)
    return 'No puede registrar una fecha anterior a la Ãºltima fecha del registro de estado del recurso';
  return null;
};

export const validateEstadoId = (values: StatusFormData): ValidationInputResult => {
  return null;
};

export const validateComentario = (values: StatusFormData): ValidationInputResult => {
  const stateName = values.recourseStatus?.filter((status) => status.id === values.statusId)[0]
    .value;
  if (
    stateName === GLOBAL_STATUS_RECOURSE.DESCARTADO ||
    stateName === GLOBAL_STATUS_RECOURSE.DESFASADO
  ) {
    if (values.comment.trim() === '') {
      return `Debe ingresar un comentario al cambiar a estado ${stateName}`;
    }
  }
  return null;
};
