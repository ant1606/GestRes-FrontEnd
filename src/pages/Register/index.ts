export { RegisterContainer as default } from './RegisterContainer';
