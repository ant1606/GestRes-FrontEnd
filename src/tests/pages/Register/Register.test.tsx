import Register from '@/pages/Register';
import { setServiceRegisterResponseSuccess } from '@/tests/mocks/register.handlers';
import { renderWithProviders } from '@/tests/utils/renderWithProvider';
import { cleanup, fireEvent, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import userEvent from '@testing-library/user-event';

describe('Test en Register', () => {
  /***
   * Casos de Test
   * [X] Debe mostrar mensajes de validación cuando se hace submit y no envia xxx dato
   * [X] Debe mostrar un mensaje de notificación si se recibe un mensaje de error del API
   * [X] Debe mostrar notificación de envio de email de verificación luego de registro de usuario satisfactorio
   */

  beforeEach(() => {
    cleanup();
    setServiceRegisterResponseSuccess(true);
  });
  afterAll(() => {
    cleanup();
    setServiceRegisterResponseSuccess(true);
  });
  test('Register  debe renderizarse', () => {
    const wrapper = renderWithProviders(<Register />);
    wrapper.unmount();
  });

  test('Debe mostrar notificación de envio de email de verificación luego de registro de usuario satisfactorio', async () => {
    const wrapper = renderWithProviders(<Register />);
    setServiceRegisterResponseSuccess(true);
    act(() => {
      fireEvent.change(wrapper.getByTestId('email'), {
        target: { value: 'testUser@email.com' }
      });
      fireEvent.change(wrapper.getByTestId('name'), { target: { value: 'test User' } });
      fireEvent.change(wrapper.getByTestId('password'), { target: { value: '123456' } });
      fireEvent.change(wrapper.getByTestId('passwordConfirmation'), {
        target: { value: '123456' }
      });
      fireEvent.click(wrapper.getByRole('button'));
    });

    await waitFor(() => {
      expect(wrapper.getByText(/enviado un link/)).toBeInTheDocument();
    });
    wrapper.unmount();
  });

  test('Debe mostrar notificación de error al recibir una respuesta de error por el API', async () => {
    const wrapper = renderWithProviders(<Register />);
    setServiceRegisterResponseSuccess(false);
    act(() => {
      fireEvent.change(wrapper.getByTestId('email'), {
        target: { value: 'testUser@email.com' }
      });
      fireEvent.change(wrapper.getByTestId('name'), { target: { value: 'test User' } });
      fireEvent.change(wrapper.getByTestId('password'), { target: { value: '123456' } });
      fireEvent.change(wrapper.getByTestId('passwordConfirmation'), {
        target: { value: '123456' }
      });
      fireEvent.click(wrapper.getByRole('button'));
    });

    await waitFor(() => {
      expect(
        wrapper.getByText(/Hubo problemas en la comunicación con el servidor/)
      ).toBeInTheDocument();
      expect(wrapper).toMatchSnapshot();
      wrapper.unmount();
    });
  });

  describe('Test de validación del formulario', () => {
    beforeEach(() => {
      cleanup();
    });

    test('Debe mostrar mensajes de validación cuando se envia el formulario sin ingresar datos', async () => {
      const wrapper = renderWithProviders(<Register />);
      const user = userEvent.setup();
      await user.click(wrapper.getByText('Registrar'));
      await waitFor(() => {
        expect(wrapper.getByText(/ingresar el nombre del usuario/)).toBeInTheDocument();
        expect(wrapper.getByText(/ingresar el email del usuario/)).toBeInTheDocument();
        expect(wrapper.getByText(/ingresar el password del usuario/)).toBeInTheDocument();
        expect(
          wrapper.getByText(/ingresar la confirmación de password del usuario/)
        ).toBeInTheDocument();
      });
      wrapper.unmount();
    });

    test('Debe mostrar mensaje de validación cuando no se ingresa nombre', async () => {
      const wrapper = renderWithProviders(<Register />);
      act(() => {
        fireEvent.change(wrapper.getByTestId('email'), {
          target: { value: 'miemail@validemail.com' }
        });
        fireEvent.change(wrapper.getByTestId('password'), { target: { value: '123456' } });
        fireEvent.change(wrapper.getByTestId('passwordConfirmation'), {
          target: { value: '123456' }
        });
        fireEvent.click(wrapper.getByText('Registrar'));
      });

      await waitFor(() => {
        expect(wrapper.getByText(/ingresar el nombre del usuario/)).toBeInTheDocument();
      });
      wrapper.unmount();
    });

    test('Debe mostrar mensaje de validación cuando no se ingresa el email', async () => {
      const wrapper = renderWithProviders(<Register />);
      act(() => {
        fireEvent.change(wrapper.getByTestId('name'), {
          target: { value: 'mi nombre' }
        });
        fireEvent.change(wrapper.getByTestId('password'), { target: { value: '123456' } });
        fireEvent.change(wrapper.getByTestId('passwordConfirmation'), {
          target: { value: '123456' }
        });
        fireEvent.click(wrapper.getByText('Registrar'));
      });

      await waitFor(() => {
        expect(wrapper.getByText(/ingresar el email del usuario/)).toBeInTheDocument();
      });
      wrapper.unmount();
    });

    test('Debe mostrar mensaje de validación cuando se ingresa formato incorrecto del email', async () => {
      const wrapper = renderWithProviders(<Register />);
      act(() => {
        fireEvent.change(wrapper.getByTestId('name'), { target: { value: 'mi nombre' } });
        fireEvent.change(wrapper.getByTestId('email'), { target: { value: 'miemail.sinformato' } });
        fireEvent.change(wrapper.getByTestId('password'), { target: { value: '123456' } });
        fireEvent.change(wrapper.getByTestId('passwordConfirmation'), {
          target: { value: '123456' }
        });
        fireEvent.click(wrapper.getByText('Registrar'));
      });

      await waitFor(() => {
        expect(wrapper.getByText(/Formato incorrecto ingresado del email/)).toBeInTheDocument();
      });
      wrapper.unmount();
    });

    test('Debe mostrar mensaje de validación cuando no se ingresa el password', async () => {
      const wrapper = renderWithProviders(<Register />);
      act(() => {
        fireEvent.change(wrapper.getByTestId('name'), { target: { value: 'mi nombre' } });
        fireEvent.change(wrapper.getByTestId('email'), { target: { value: 'mi@email.com' } });
        fireEvent.change(wrapper.getByTestId('passwordConfirmation'), {
          target: { value: '123456' }
        });
        fireEvent.click(wrapper.getByText('Registrar'));
      });

      await waitFor(() => {
        expect(wrapper.getByText(/ingresar el password del usuario/)).toBeInTheDocument();
      });
      wrapper.unmount();
    });

    test('Debe mostrar mensaje de validación cuando no se ingresa el password confirmation', async () => {
      const wrapper = renderWithProviders(<Register />);
      act(() => {
        fireEvent.change(wrapper.getByTestId('name'), { target: { value: 'mi nombre' } });
        fireEvent.change(wrapper.getByTestId('email'), { target: { value: 'mi@email.com' } });
        fireEvent.change(wrapper.getByTestId('password'), { target: { value: '123456' } });
        fireEvent.click(wrapper.getByText('Registrar'));
      });

      await waitFor(() => {
        expect(
          wrapper.getByText(/ingresar la confirmación de password del usuario/)
        ).toBeInTheDocument();
      });
      wrapper.unmount();
    });

    test('Debe mostrar mensaje de validación cuando el password Confirmation no es igual al password', async () => {
      const wrapper = renderWithProviders(<Register />);
      act(() => {
        fireEvent.change(wrapper.getByTestId('name'), { target: { value: 'mi nombre' } });
        fireEvent.change(wrapper.getByTestId('email'), { target: { value: 'mi@email.com' } });
        fireEvent.change(wrapper.getByTestId('password'), { target: { value: '123456' } });
        fireEvent.change(wrapper.getByTestId('passwordConfirmation'), {
          target: { value: '12345678910' }
        });
        fireEvent.click(wrapper.getByText('Registrar'));
      });

      await waitFor(() => {
        expect(wrapper.getByText(/Las contraseñas ingresadas no son iguales/)).toBeInTheDocument();
      });
      wrapper.unmount();
    });
  });
});
