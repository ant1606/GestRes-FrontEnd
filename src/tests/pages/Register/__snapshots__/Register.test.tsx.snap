// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test en Register > Debe mostrar notificación de error al recibir una respuesta de error por el API 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="swal2-shown swal2-height-auto"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="flex flex-col h-screen"
      >
        <div
          class="bg-gray-800 min-h-[5rem]"
        />
        <div
          class="bg-gray-200 h-full flex justify-center align-center"
        >
          <div
            class="flex flex-col justify-center align-center gap-6 min-w-[30rem]"
          >
            <p
              class="text-4xl leading-10 font-bold text-center"
            >
              Registro de Usuario
            </p>
            <form
              class="flex flex-col justify-center gap-4"
            >
              <div
                class="flex gap-3 items-center"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,19.2C9.5,19.2 7.29,17.92 6,16C6.03,14 10,12.9 12,12.9C14,12.9 17.97,14 18,16C16.71,17.92 14.5,19.2 12,19.2M12,5A3,3 0 0,1 15,8A3,3 0 0,1 12,11A3,3 0 0,1 9,8A3,3 0 0,1 12,5M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2Z"
                    style="fill: currentColor;"
                  />
                </svg>
                <div
                  class="flex flex-col relative my-3 grow"
                >
                  <div
                    class=" 
        relative "
                  >
                    <input
                      autocomplete="off"
                      class="border-gray-900
            px-3 py-1  border-b-2 bg-white peer w-full text-base transition-colors h-8 focus:outline-none"
                      data-testid="name"
                      id="name"
                      name="name"
                      placeholder=" "
                      type="text"
                      value="test User"
                    />
                    <label
                      class="text-gray-600 peer-focus:text-gray-900 
          absolute left-0 -top-4 text-xs cursor-text transition-all px-0 peer-placeholder-shown:top-1 peer-placeholder-shown:text-base peer-placeholder-shown:px-3
          peer-focus:-top-4 peer-focus:text-xs  peer-focus:px-0"
                      for="name"
                    >
                      Nombre
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="flex gap-3 items-center"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20,8L12,13L4,8V6L12,11L20,6M20,4H4C2.89,4 2,4.89 2,6V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V6C22,4.89 21.1,4 20,4Z"
                    style="fill: currentColor;"
                  />
                </svg>
                <div
                  class="flex flex-col relative my-3 grow"
                >
                  <div
                    class=" 
        relative "
                  >
                    <input
                      autocomplete="off"
                      class="border-gray-900
            px-3 py-1  border-b-2 bg-white peer w-full text-base transition-colors h-8 focus:outline-none"
                      data-testid="email"
                      id="email"
                      name="email"
                      placeholder=" "
                      type="text"
                      value="testUser@email.com"
                    />
                    <label
                      class="text-gray-600 peer-focus:text-gray-900 
          absolute left-0 -top-4 text-xs cursor-text transition-all px-0 peer-placeholder-shown:top-1 peer-placeholder-shown:text-base peer-placeholder-shown:px-3
          peer-focus:-top-4 peer-focus:text-xs  peer-focus:px-0"
                      for="email"
                    >
                      Email
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="flex gap-3 items-center"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,17A2,2 0 0,0 14,15C14,13.89 13.1,13 12,13A2,2 0 0,0 10,15A2,2 0 0,0 12,17M18,8A2,2 0 0,1 20,10V20A2,2 0 0,1 18,22H6A2,2 0 0,1 4,20V10C4,8.89 4.9,8 6,8H7V6A5,5 0 0,1 12,1A5,5 0 0,1 17,6V8H18M12,3A3,3 0 0,0 9,6V8H15V6A3,3 0 0,0 12,3Z"
                    style="fill: currentColor;"
                  />
                </svg>
                <div
                  class="flex flex-col relative my-3 grow"
                >
                  <div
                    class=" 
        relative "
                  >
                    <input
                      autocomplete="off"
                      class="border-gray-900
            px-3 py-1  border-b-2 bg-white peer w-full text-base transition-colors h-8 focus:outline-none"
                      data-testid="password"
                      id="password"
                      name="password"
                      placeholder=" "
                      type="password"
                      value="123456"
                    />
                    <label
                      class="text-gray-600 peer-focus:text-gray-900 
          absolute left-0 -top-4 text-xs cursor-text transition-all px-0 peer-placeholder-shown:top-1 peer-placeholder-shown:text-base peer-placeholder-shown:px-3
          peer-focus:-top-4 peer-focus:text-xs  peer-focus:px-0"
                      for="password"
                    >
                      Password
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="flex gap-3 items-center"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13C19.34 13 19.67 13.04 20 13.09V10C20 8.9 19.11 8 18 8H17V6C17 3.24 14.76 1 12 1S7 3.24 7 6V8H6C4.9 8 4 8.89 4 10V20C4 21.11 4.89 22 6 22H13.81C13.3 21.12 13 20.1 13 19C13 15.69 15.69 13 19 13M9 6C9 4.34 10.34 3 12 3S15 4.34 15 6V8H9V6M12 17C10.9 17 10 16.11 10 15S10.9 13 12 13C13.1 13 14 13.89 14 15C14 16.11 13.11 17 12 17M22.5 17.25L17.75 22L15 19L16.16 17.84L17.75 19.43L21.34 15.84L22.5 17.25Z"
                    style="fill: currentColor;"
                  />
                </svg>
                <div
                  class="flex flex-col relative my-3 grow"
                >
                  <div
                    class=" 
        relative "
                  >
                    <input
                      autocomplete="off"
                      class="border-gray-900
            px-3 py-1  border-b-2 bg-white peer w-full text-base transition-colors h-8 focus:outline-none"
                      data-testid="passwordConfirmation"
                      id="passwordConfirmation"
                      name="passwordConfirmation"
                      placeholder=" "
                      type="password"
                      value="123456"
                    />
                    <label
                      class="text-gray-600 peer-focus:text-gray-900 
          absolute left-0 -top-4 text-xs cursor-text transition-all px-0 peer-placeholder-shown:top-1 peer-placeholder-shown:text-base peer-placeholder-shown:px-3
          peer-focus:-top-4 peer-focus:text-xs  peer-focus:px-0"
                      for="passwordConfirmation"
                    >
                      Confirmar Password
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="flex"
              >
                <button
                  class="bg-gray-900 rounded-xl text-white py-2 px-5 text-2xl font-medium
        hover:bg-gray-800 grow"
                  type="submit"
                >
                  Registrar
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          class="bg-gray-900 min-h-[5rem]"
        />
      </div>
    </div>
    <div
      class="swal2-container swal2-center swal2-backdrop-show"
    >
      <div
        aria-describedby="swal2-html-container"
        aria-labelledby="swal2-title"
        aria-live="assertive"
        aria-modal="true"
        class="swal2-popup swal2-modal swal2-icon-error"
        role="dialog"
        style="opacity: 0 !important; display: grid;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="swal2-close"
          style="display: none;"
          type="button"
        >
          ×
        </button>
        <ul
          class="swal2-progress-steps"
          style="display: none;"
        />
        <div
          class="swal2-icon swal2-error swal2-icon-show"
          style="display: flex;"
        >
          <span
            class="swal2-x-mark"
          >
            
    
            <span
              class="swal2-x-mark-line-left"
            />
            
    
            <span
              class="swal2-x-mark-line-right"
            />
            
  
          </span>
          

        </div>
        <img
          class="swal2-image"
          style="display: none;"
        />
        <h2
          class="swal2-title"
          id="swal2-title"
          style="display: block;"
        >
          Sistema
        </h2>
        <div
          class="swal2-html-container"
          id="swal2-html-container"
          style="display: block;"
        >
          Hubo problemas en la comunicación con el servidor
        </div>
        <input
          class="swal2-input"
          style="display: none;"
        />
        <input
          class="swal2-file"
          style="display: none;"
          type="file"
        />
        <div
          class="swal2-range"
          style="display: none;"
        >
          <input
            type="range"
          />
          <output />
        </div>
        <select
          class="swal2-select"
          style="display: none;"
        />
        <div
          class="swal2-radio"
          style="display: none;"
        />
        <label
          class="swal2-checkbox"
          for="swal2-checkbox"
          style="display: none;"
        >
          <input
            type="checkbox"
          />
          <span
            class="swal2-label"
          />
        </label>
        <textarea
          class="swal2-textarea"
          style="display: none;"
        />
        <div
          class="swal2-validation-message"
          id="swal2-validation-message"
          style="display: none;"
        />
        <div
          class="swal2-actions"
          style="display: flex;"
        >
          <div
            class="swal2-loader"
          />
          <button
            aria-label=""
            class="swal2-confirm swal2-styled"
            style="display: inline-block;"
            type="button"
          >
            OK
          </button>
          <button
            aria-label=""
            class="swal2-deny swal2-styled"
            style="display: none;"
            type="button"
          >
            No
          </button>
          <button
            aria-label=""
            class="swal2-cancel swal2-styled"
            style="display: none;"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div
          class="swal2-footer"
          style="display: none;"
        />
        <div
          class="swal2-timer-progress-bar-container"
        >
          <div
            class="swal2-timer-progress-bar"
            style="display: none;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="flex flex-col h-screen"
    >
      <div
        class="bg-gray-800 min-h-[5rem]"
      />
      <div
        class="bg-gray-200 h-full flex justify-center align-center"
      >
        <div
          class="flex flex-col justify-center align-center gap-6 min-w-[30rem]"
        >
          <p
            class="text-4xl leading-10 font-bold text-center"
          >
            Registro de Usuario
          </p>
          <form
            class="flex flex-col justify-center gap-4"
          >
            <div
              class="flex gap-3 items-center"
            >
              <svg
                role="presentation"
                style="width: 1.5rem; height: 1.5rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,19.2C9.5,19.2 7.29,17.92 6,16C6.03,14 10,12.9 12,12.9C14,12.9 17.97,14 18,16C16.71,17.92 14.5,19.2 12,19.2M12,5A3,3 0 0,1 15,8A3,3 0 0,1 12,11A3,3 0 0,1 9,8A3,3 0 0,1 12,5M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2Z"
                  style="fill: currentColor;"
                />
              </svg>
              <div
                class="flex flex-col relative my-3 grow"
              >
                <div
                  class=" 
        relative "
                >
                  <input
                    autocomplete="off"
                    class="border-gray-900
            px-3 py-1  border-b-2 bg-white peer w-full text-base transition-colors h-8 focus:outline-none"
                    data-testid="name"
                    id="name"
                    name="name"
                    placeholder=" "
                    type="text"
                    value="test User"
                  />
                  <label
                    class="text-gray-600 peer-focus:text-gray-900 
          absolute left-0 -top-4 text-xs cursor-text transition-all px-0 peer-placeholder-shown:top-1 peer-placeholder-shown:text-base peer-placeholder-shown:px-3
          peer-focus:-top-4 peer-focus:text-xs  peer-focus:px-0"
                    for="name"
                  >
                    Nombre
                  </label>
                </div>
              </div>
            </div>
            <div
              class="flex gap-3 items-center"
            >
              <svg
                role="presentation"
                style="width: 1.5rem; height: 1.5rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20,8L12,13L4,8V6L12,11L20,6M20,4H4C2.89,4 2,4.89 2,6V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V6C22,4.89 21.1,4 20,4Z"
                  style="fill: currentColor;"
                />
              </svg>
              <div
                class="flex flex-col relative my-3 grow"
              >
                <div
                  class=" 
        relative "
                >
                  <input
                    autocomplete="off"
                    class="border-gray-900
            px-3 py-1  border-b-2 bg-white peer w-full text-base transition-colors h-8 focus:outline-none"
                    data-testid="email"
                    id="email"
                    name="email"
                    placeholder=" "
                    type="text"
                    value="testUser@email.com"
                  />
                  <label
                    class="text-gray-600 peer-focus:text-gray-900 
          absolute left-0 -top-4 text-xs cursor-text transition-all px-0 peer-placeholder-shown:top-1 peer-placeholder-shown:text-base peer-placeholder-shown:px-3
          peer-focus:-top-4 peer-focus:text-xs  peer-focus:px-0"
                    for="email"
                  >
                    Email
                  </label>
                </div>
              </div>
            </div>
            <div
              class="flex gap-3 items-center"
            >
              <svg
                role="presentation"
                style="width: 1.5rem; height: 1.5rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,17A2,2 0 0,0 14,15C14,13.89 13.1,13 12,13A2,2 0 0,0 10,15A2,2 0 0,0 12,17M18,8A2,2 0 0,1 20,10V20A2,2 0 0,1 18,22H6A2,2 0 0,1 4,20V10C4,8.89 4.9,8 6,8H7V6A5,5 0 0,1 12,1A5,5 0 0,1 17,6V8H18M12,3A3,3 0 0,0 9,6V8H15V6A3,3 0 0,0 12,3Z"
                  style="fill: currentColor;"
                />
              </svg>
              <div
                class="flex flex-col relative my-3 grow"
              >
                <div
                  class=" 
        relative "
                >
                  <input
                    autocomplete="off"
                    class="border-gray-900
            px-3 py-1  border-b-2 bg-white peer w-full text-base transition-colors h-8 focus:outline-none"
                    data-testid="password"
                    id="password"
                    name="password"
                    placeholder=" "
                    type="password"
                    value="123456"
                  />
                  <label
                    class="text-gray-600 peer-focus:text-gray-900 
          absolute left-0 -top-4 text-xs cursor-text transition-all px-0 peer-placeholder-shown:top-1 peer-placeholder-shown:text-base peer-placeholder-shown:px-3
          peer-focus:-top-4 peer-focus:text-xs  peer-focus:px-0"
                    for="password"
                  >
                    Password
                  </label>
                </div>
              </div>
            </div>
            <div
              class="flex gap-3 items-center"
            >
              <svg
                role="presentation"
                style="width: 1.5rem; height: 1.5rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13C19.34 13 19.67 13.04 20 13.09V10C20 8.9 19.11 8 18 8H17V6C17 3.24 14.76 1 12 1S7 3.24 7 6V8H6C4.9 8 4 8.89 4 10V20C4 21.11 4.89 22 6 22H13.81C13.3 21.12 13 20.1 13 19C13 15.69 15.69 13 19 13M9 6C9 4.34 10.34 3 12 3S15 4.34 15 6V8H9V6M12 17C10.9 17 10 16.11 10 15S10.9 13 12 13C13.1 13 14 13.89 14 15C14 16.11 13.11 17 12 17M22.5 17.25L17.75 22L15 19L16.16 17.84L17.75 19.43L21.34 15.84L22.5 17.25Z"
                  style="fill: currentColor;"
                />
              </svg>
              <div
                class="flex flex-col relative my-3 grow"
              >
                <div
                  class=" 
        relative "
                >
                  <input
                    autocomplete="off"
                    class="border-gray-900
            px-3 py-1  border-b-2 bg-white peer w-full text-base transition-colors h-8 focus:outline-none"
                    data-testid="passwordConfirmation"
                    id="passwordConfirmation"
                    name="passwordConfirmation"
                    placeholder=" "
                    type="password"
                    value="123456"
                  />
                  <label
                    class="text-gray-600 peer-focus:text-gray-900 
          absolute left-0 -top-4 text-xs cursor-text transition-all px-0 peer-placeholder-shown:top-1 peer-placeholder-shown:text-base peer-placeholder-shown:px-3
          peer-focus:-top-4 peer-focus:text-xs  peer-focus:px-0"
                    for="passwordConfirmation"
                  >
                    Confirmar Password
                  </label>
                </div>
              </div>
            </div>
            <div
              class="flex"
            >
              <button
                class="bg-gray-900 rounded-xl text-white py-2 px-5 text-2xl font-medium
        hover:bg-gray-800 grow"
                type="submit"
              >
                Registrar
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="bg-gray-900 min-h-[5rem]"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
